### PR TITLE
:sparkles: add support for basic PDF operations

### PR DESCRIPTION
**Pull request by sebastianMindee**
## Description

* :sparkles: add support for PDF page cutting
* :recycle: update workflows
* :wrench: add license check workflow

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires a change to the official Guide documentation.

### CODE REVIEW

File: src/Input/LocalInputSource.php [hunk 1]
```diff
@@ -93,42 +97,139 @@ public function isPDF(): bool
...
      * Counts the amount of pages in a PDF.
-     * To be implemented.
      *
-     * @return void
+     * @return integer
+     * @throws MindeePDFException Throws if the source pdf can't be properly processed.
+     * @throws MindeeSourceException Throws if the source isn't a pdf.
      */
-    public function countDocPages() // TODO: add PDF lib
+    public function countDocPages(): int
```
Comment (ianardee) [inline src/Input/LocalInputSource.php:105]: It doesn't look like the page count is returned, unless I'm missing something.

File: tests/Input/LocalInputSourceTest.php [hunk 2]
```diff
@@ -49,41 +59,99 @@ public function testPDFReadContents()
...
+            for ($pageNumber = 0; $pageNumber < $pageCountBasePdf; $pageNumber++) {
+                $cutPdf->setSourceFile($this->fileTypesDir . "pdf/multipage_cut-" . count($indexes) . ".pdf");
+                $basePdf->setSourceFile($inputObj->fileObject->getFilename());
+                $cutPdf->AddPage();
+                $cutPdf->useTemplate($cutPdf->importPage($pageNumber + 1));
+                $basePdf->AddPage();
+                $basePdf->useTemplate($basePdf->importPage($pageNumber + 1));
+                // Note: comparing extracted page bytes content turns out to be reliable when using FPDF.
+                // This will be left here until a better solution is found within the limitations of licensing.
+                //                $this->assertEquals($cutPdf->Output('', 'S'), $basePdf->Output('', 'S'));
```
Comment (ianardee) [inline tests/Input/LocalInputSourceTest.php:92]: not sure I understand your comment, shouldn't the assertion be executed, if it's reliable?

**Review by ianardee — CHANGES_REQUESTED** [review]

File: src/Input/LocalInputSource.php (hunk 1 above)
Comment (sebastianMindee) [reply src/Input/LocalInputSource.php:105]: Yeah the syntax is [hella weird](https://manuals.setasign.com/fpdi-manual/v1/the-fpdi-class/#index-3-1), but it does return the page count.

File: tests/Input/LocalInputSourceTest.php (hunk 2 above)
Comment (sebastianMindee) [reply tests/Input/LocalInputSourceTest.php]: typo, whoops

**Review by ianardee — APPROVED** [review]

File: src/Input/LocalInputSource.php (hunk 1 above)
Comment (ianardee) [reply src/Input/LocalInputSource.php:105]: Such API, very clean. Much logic.

**Pull request merged**